### DISCUSSION TITLE
CGDK2 shows temperature and humidity of 0,0 after update from 4.7 to 4.8/4.9

### DISCUSSION THREAD

**Discussion by TheZoker** [pvvx/ATC_MiThermometer#588] [Q&A]
After updating my CGDK2 from 4.7 (which works without any issue) to 4.8 or the new 4.9 beta, the device only shows temperature of 0.0 and also for humidity. Do I need to set something or is this a bug? 

After downgrading to 4.7 again, the temperature is shown correct again.

**Comment by pvvx** [comment]
There are no special changes for CGDK2 released by Qingping with a defective sensor since version 4.7. CGDK2 with a normal sensor everything works.

**Reply by TheZoker** [reply]
Strange. I now flashed the firmware 2 times from 4.7 to 4.8 and the sensor is always broke. 
When I flash it back to 4.7 it works. 
Is there something else I can test?

**Reply by pvvx** [reply]
Qingping released a variant with a defective sensor chip to the market. Adaptation and bug fixing were made for such batches of sensors. Why it does not work now is unknown.

**Reply by TheZoker** [reply]
But nothing changes there from 4.6 to 4.7, right? 
So when it works with 4.6 it should also work with 4.7?

**Reply by pvvx** [reply]
But the sensor has no identifier and some commands do not work at its address on the I2C bus. Some commands receive a response when requesting the reserved address "0". Otherwise, this is a typical SHTV3 sensor, but with a defect.

https://github.com/pvvx/ATC_MiThermometer/issues/273

The reasons why and because of what addressing errors occur in it are unknown.
It is not possible to identify, distinguish from other types of sensors, since it does not respond to a typical ID request at its address on the I2C bus.

**Reply by pvvx** [reply]
> But nothing changes there from 4.6 to 4.7, right?

Yes.

> So when it works with 4.6 it should also work with 4.7?

Not necessarily.
Adding autodetection for other sensors changes the polling start sequence. 
This sensor may not like any pulses on the I2C bus and will start working differently... 
To correct this, complete documentation for this sensor version is required. With the ability to bypass oddities/bugs in the crystal.

And I don't have such CGDK2.
To solve this, I have already bought many of these CGDK2, but all of them came with working sensors.

I can't buy other options outside my country - all accounts there are blocked by some sanctions. That is, your rulers simply stole everything.

**Reply by TheZoker** [reply]
Thanks for the explanation!

I bought these here:
https://www.aliexpress.com/item/1005005900568131.html

Are you able to order these?

**Reply by pvvx** [reply]
![image](https://github.com/user-attachments/assets/8d4e7dec-4ccd-4c64-b52b-36c9b005115e)

---

I'm not going to buy every CGDK2 on AliExpress in the hopes that I'll get one with a bad sensor. 

![image](https://github.com/user-attachments/assets/3c1e995a-4657-4db4-ba7f-aeea04dc640f)
